src/Sulu/Component/HttpCache/HttpCache.php

```diff
@@ -11,27 +11,54 @@
 
 namespace Sulu\Component\HttpCache;
 
-use Symfony\Bundle\FrameworkBundle\HttpCache\HttpCache as HttpCacheAbstract;
+use FOS\HttpCache\SymfonyCache\EventDispatchingHttpCache;
+use FOS\HttpCache\SymfonyCache\UserContextSubscriber;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpCache\Esi;
+use Symfony\Component\HttpKernel\HttpCache\Store;
 use Symfony\Component\HttpKernel\HttpKernelInterface;
 
 /**
  * Sulu HttpCache - Lookups a valid Response from the cache for the given request
  * or forwards the Request to the backend and stores the Response in the cache.
  */
-class HttpCache extends HttpCacheAbstract
+class HttpCache extends EventDispatchingHttpCache
 {
     const HEADER_REVERSE_PROXY_TTL = 'X-Reverse-Proxy-TTL';
 
+    const USER_HASH_URI = '/_user_context';
+
+    const USER_HASH_HEADER = 'X-User-Context-Hash';
+
+    const SESSION_NAME_PREFIX = 'user-context';
+
+    public function __construct(HttpKernelInterface $kernel, $hasUserContext = false, $cacheDir = null)
+    {
+        parent::__construct(
+            $kernel,
+            new Store($cacheDir ?: $kernel->getCacheDir().'/http_cache'),
+            new Esi(),
+            ['debug' => $kernel->isDebug()]
+        );
+
+        if ($hasUserContext) {
```

Comment:
Since our HTTP cache implementation is already doing quite a lot I would like to add some kind of test to it. Especially because we are also depending on an external library.